x/auth/mock/simulate_block.go

```diff
@@ -11,13 +11,52 @@ import (
 	crypto "github.com/tendermint/go-crypto"
 
 	abci "github.com/tendermint/abci/types"
+	"math/rand"
 )
 
 var chainID = "" // TODO
 
+// Set genesis accounts with random coin values using the provided addresses and
+// coin denominations
+func RandomSetGenesis(r *rand.Rand, app *App, addrs []sdk.Address, denoms []string) {
+	accts := make([]auth.Account, len(addrs), len(addrs))
+	maxNumCoins := 2 << 50
```

Comment:
Interesting, why `2 << 50`?

Reply:
That was based off of the conversation we had for the max value we expect to be in an int64, to avoid overflow concerns in most cases. Since we're switching to int256, I'll switch to that once thats in develop.